Refactor User Registration and Login Endpoints

HijoDeHonor opened the pull request:
This pull request includes the following changes:

### Refactor User Registration Endpoint (Branch 60)
- Refactored the user registration endpoint to utilize a dedicated service.
- Integrated the endpoint with the dependency injection container.
- Added unit tests to ensure the functionality of the user registration service.
- Renamed files for better clarity and consistency.
- Removed the 'services' path for improved project structure.

### Refactor Login Endpoint (Branch 62)
- Refactored the login endpoint to utilize a dedicated service.
- Integrated the login endpoint with the dependency injection container.
- Updated the unit tests for the login endpoint.
- Renamed variables to improve code readability.

These changes improve the modularity, maintainability, and readability of the codebase.

HijoDeHonor committed:
refactor: refactor user registration endpoint to use service, add unit tests

Refactored the user registration endpoint to utilize a dedicated service for user creation. Added unit tests to ensure the functionality of the service.

HijoDeHonor committed:
refactor: fix import issues in tests, rename files, remove services path

- Corrected import issues in test files
- Renamed files for better clarity and consistency
- Removed the 'services' path for improved project structure

HijoDeHonor committed:
refactor: refactor login endpoint to use service and DI, update tests

- Refactored the login endpoint to utilize a dedicated service and integrated it with the dependency injection container
- Updated the unit tests for the login endpoint
- Renamed variables to improve code readability

PiewDev commented on users/userService.js:
```diff
@@ -0,0 +1,37 @@
+  }
+
+  create = async (userName, password) => {
+    const userValidated = userValidate(userName, password);
```
here userValidation seems like a better option to me. because its more legible in if statement.

PiewDev commented on users/userService.js:
```diff
@@ -0,0 +1,37 @@
+  };
+
+  login = async (userName, password) => {
+    const validation = userValidate(userName, password);
```
It is not 100% necessary to keep the same name that you used in the validation above but it is a good practice

PiewDev replied:
Maybe you could make a validation function:
`validateUser(username,password)`

PiewDev replied:
On service level.

PiewDev commented on users/userService.js:
```diff
@@ -0,0 +1,37 @@
+    if (findUser[0].password !== password) {
+      throw new InvalidLoginError(INVALID_LOGIN, USERS);
+    }
+    const token = jwtCreator(findUser[0]);
```
This is the controller's responsibility

PiewDev commented on users/userValidate.js:
The name must be userValidator

PiewDev commented on users/userService.js:
```diff
@@ -0,0 +1,37 @@
+    if (!validation.success) {
+      throw new InvalidDataError(INVALID_DATA, USERS);
+    }
+    const findUser = await this.userRepository.getBy({ userName });
```
findUser is not a good name. Maybe just user is better.

PiewDev replied:
try:
`const [user] = await this.userRepository.getBy({ userName }) || [];`

PiewDev commented on users/userService.js:
```diff
@@ -0,0 +1,37 @@
+      throw new InvalidLoginError(INVALID_LOGIN, USERS);
+    }
+    const token = jwtCreator(findUser[0]);
+    return { findUser, token };
```
In this case it does not make sense to return an array with a single element, as it will always be the case in this case to return the element directly.

PiewDev reviewed (changes requested):
There are some changes to make

HijoDeHonor committed:
refactor:
Refactored data validation logic and updated corresponding tests.
- Added default values for environment variables to ensure robust configuration.
- Corrected responsibilities between the controller and the service for better separation of concerns.
